x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java

```diff
@@ -182,18 +187,19 @@ public void createResponse(String docId,
                                R response,
                                ActionListener<IndexResponse> listener) throws IOException {
         try {
-            // TODO: Integrate with circuit breaker
-            final XContentBuilder source = XContentFactory.jsonBuilder()
+            final ReleasableBytesStreamOutput buffer = new ReleasableBytesStreamOutput(0, bigArrays.withCircuitBreaking());
+            listener = ActionListener.runBefore(listener, buffer::close);
```

Comment:
Small comment, maybe this could be `source::close`.

Reply:
May be on line 202 instead of `buffer.bytes()` , we can use `BytesReference.bytes(source)` will be automatically close `source`. 

This part of the code is quite complex, there are a lot of hidden things happening inside.

Reply:
Ok, I added a comment in https://github.com/elastic/elasticsearch/pull/73862/commits/b35fd3b1cef81484cc79cad69af0fc80a881ecc9 and close the source in https://github.com/elastic/elasticsearch/pull/73862/commits/345c4d191d02b191d96c9f3f2424c5e1cc368f96